File: java-hfsframework-primefaces/hfsframework-admin/src/main/java/br/com/hfsframework/admin/data/AdmFuncionalidadeRepository.java
/**
 * <p><b>HFS Framework</b></p>
 * @author Henrique Figueiredo de Souza
 * @version 1.0
 * @since 2017
 */
package br.com.hfsframework.admin.data;

import java.util.List;

import org.apache.deltaspike.data.api.EntityRepository;
import org.apache.deltaspike.data.api.Query;
import org.apache.deltaspike.data.api.Repository;

import br.com.hfsframework.admin.model.AdmFuncionalidade;
import br.com.hfsframework.admin.model.AdmMenu;
import br.com.hfsframework.admin.model.AdmPagina;
import br.com.hfsframework.admin.model.AdmPerfil;

/**
 * The Interface AdmFuncionalidadeRepository.
 */
@Repository(forEntity = AdmFuncionalidade.class)
public interface AdmFuncionalidadeRepository extends EntityRepository<AdmFuncionalidade, Long> {

	/**
	 * Gets the descricao.
	 *
	 * @param id
	 *            the id
	 * @return the descricao
	 */
	@Query(named = "AdmFuncionalidade.getDescricaoById")
	String getDescricao(Long id);
	
	/**
	 * Count novo.
	 *
	 * @param novo
	 *            the novo
	 * @return the long
	 */
	@Query(named = "AdmFuncionalidade.countNovo")
	Long countNovo(String novo);

	/**
	 * Count antigo novo.
	 *
	 * @param antigo
	 *            the antigo
	 * @param novo
	 *            the novo
	 * @return the long
	 */
	@Query(named = "AdmFuncionalidade.countAntigo")
	Long countAntigo(String antigo, String novo);
	
	/**
	 * Find perfis por funcionalidade.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @return the list
	 */
	@Query(named = "AdmFuncionalidade.findPerfisPorFuncionalidade")
	List<AdmPerfil> findPerfisPorFuncionalidade(AdmFuncionalidade funcionalidade);

	/**
	 * Find paginas por funcionalidade.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @return the list
	 */
	@Query(named = "AdmFuncionalidade.findPaginasPorFuncionalidade")
	List<AdmPagina> findPaginasPorFuncionalidade(AdmFuncionalidade funcionalidade);

	/**
	 * Find menus por funcionalidade.
	 *
	 * @param funcionalidade
	 *            the funcionalidade
	 * @return the list
	 */
	@Query(named = "AdmFuncionalidade.findMenusPorFuncionalidade")
	List<AdmMenu> findMenusPorFuncionalidade(AdmFuncionalidade funcionalidade);

	/**
	 * Find perfis por URL.
	 *
	 * @param url
	 *            the url
	 * @return the list
	 */
	@Query(named = "AdmFuncionalidade.findPerfisPorURL")
	List<AdmPerfil> findPerfisPorURL(String url);
	
}

File: csharp-hfsframework/hfsframework-admin/hfsframework.admin.data/AdmFuncionalidadeRepository.cs
using System.Collections.Generic;
using hfsframework.hfsbase;
using hfsframework.admin.model;
using hfsframework.util;
using NHibernate;

namespace hfsframework.admin.data
{
    public class AdmFuncionalidadeRepository: EntityRepository<AdmFuncionalidade, long>
    {
        public override string getDescricao(long id)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.getDescricaoById");
                q.SetInt64(0, long.Parse(id.ToString()));
                return q.UniqueResult<string>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public override bool existeNovo(string novo)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.countNovo");
                q.SetString(0, novo.ToLower());
                return (q.UniqueResult<long>() > 0);
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public override bool existeAntigo(long id, string novo)
        {
            string antigo = getDescricao(id);

            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.countAntigo");
                q.SetString(0, antigo.ToLower());
                q.SetString(1, novo.ToLower());
                return (q.UniqueResult<long>() > 0);
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmPerfil> findPerfisPorFuncionalidade(AdmFuncionalidade funcionalidade)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.findPerfisPorFuncionalidade");
                q.SetEntity(0, funcionalidade);
                return q.List<AdmPerfil>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmPagina> findPaginasPorFuncionalidade(AdmFuncionalidade funcionalidade)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.findPaginasPorFuncionalidade");
                q.SetEntity(0, funcionalidade);
                return q.List<AdmPagina>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmMenu> findMenusPorFuncionalidade(AdmFuncionalidade funcionalidade)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.findMenusPorFuncionalidade");
                q.SetEntity(0, funcionalidade);
                return q.List<AdmMenu>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

        public IList<AdmPerfil> findPerfisPorURL(string url)
        {
            ISession session = NHibernateHelper.GetCurrentSession();
            try
            {
                IQuery q = session.GetNamedQuery("AdmFuncionalidade.findPerfisPorURL");
                q.SetString(0, url);
                q.SetString(1, url);
                return q.List<AdmPerfil>();
            }
            finally
            {
                NHibernateHelper.CloseSession();
            }
        }

    }
}
